### ISSUE TITLE
Plotting Morphology can be very slow

### ISSUE THREAD

**Issue by thesamovar** [brian-team/brian2tools#13]
e.g. try plotting ``c11563.CNG.swc`` from neuromorpho.org. It didn't finish after several runs. I think this is because it's doing a lot of work constructing Morphology objects? Not sure. In any case, the following code plots it very quickly and efficiently directly from the swc file so it should be possible to do this from a Morphology object too:

```
def plot_swc(fname, showtype=(True,)*4):
    # Load SWC file directly (Morphology is too slow)
    # index, type, x, y, z, radius, parent
    i, t, x, y, z, r, p = loadtxt(fname).T
    i = array(i, dtype=int)-1 # indices start at 1
    t = array(t, dtype=int)-1 # types start at 1
    p = array(p, dtype=int)
    p[p>0] -= 1 # parent indices too, but -1 is special (root node)
    # Create and plot a LineCollection
    segs = [((x[i], y[i]), (x[p[i]], y[p[i]])) for i in xrange(len(t)) if p[i]>=0 and showtype[t[i]]]
    c = LineCollection(segs, linewidths=1, colors='k', linestyle='solid')
    gca().add_collection(c)
    gca().autoscale_view(True,True,True)    
    xticks([])
    yticks([])
    gca().set_frame_on(False)
    gca().set_aspect('equal', 'datalim')    
```

Here's the swc file zipped: [c11563.CNG.swc.zip](https://github.com/brian-team/brian2tools/files/440374/c11563.CNG.swc.zip)

**Comment by mstimberg** [comment]
This is something I noticed previously, I thought I had already opened an issue about it somewhere... The plotting time is highly superlinear, it goes from "slow" for a morphology with ~80 sections to "unusable" with ~250 sections (as in your example). The plotting code in `brian2tools` was written in a lazy and inefficient way (it calls `plot` once for every section), but this is not actually the reason, the real bottleneck is getting the coordinates of deeply nested sections in a `Morphology`. I opened an issue about this here: brian-team/brian2#741 and I'll open a PR with a fix soon.
